src/guides/optimism-bridge-widget/index.md

```diff
@@ -0,0 +1,1877 @@
+
+- [`contracts/optimism/GreeterL2.sol`](https://github.com/truffle-box/optimism-bridge-box/blob/main/contracts/optimism/GreeterL2.sol)
+
+  This contract will be deployed on Optimism and controls a Greeter on Ethereum.
+
+- `truffle-config.js` and `truffle-config.ovm.js`
+  
+  In order to specify which chain to deploy to, we introduce a second config - `truffle-config.ovm.js` - with its own build path, contract directory, and networks. By default, `truffle migrate` will use `truffle-config.js`. When we want to deploy our Optimism contracts, however, we can specify `--config truffle-config.ovm.js` so that we properly deploy the correct contracts to the relevant networks.
+
+- `migrations`
+
+  In this box, we have four different migrations files. `1_deploy_L1_contracts.js` deploys the Ethereum contract `GreeterL1.sol`. `2_deploy_L2_contracts.js` deploys the Optimism contract `GreeterL2.sol`. The remaining 2 migrations, `3_set_L2_greeting.js` and `4_set_L1_greeting.js` actually don't deploy contracts - they call contract functions to send messages across L1 and L2.
+  
+  Normally, `3_set_L2_greeting.js` and `4_set_L1_greeting.js` would just go in a separate Truffle script since it's just sending messages rather than deploying a contract. However, in this case, we wanted to illustrate how with Truffle, you can write ordered, incremental steps with migrations to end in a desired state. You can see how this is done in `scripts/goerli_bridge_message.mjs`.
+
+- `scripts/goerli_bridge_message.mjs`
+
+  `npm run deploy` runs this script - which is actually a series of `truffle migrate` calls. This automates the process of compiling contracts, running migrations, and sending messages across each side of the bridge. Note the `--f 1 --to 1`, `--f 2 --to 2`, `--f 3 --to 3`, and `--f 4 --to 4` flags. This is how we tell Truffle which migrations to run provided we prefix the migration script's name with the appropriate number.
+
+- `scripts/goerli_bridge_value.mjs`
+
+  This script automates the process of compiling contracts, running migrations, and sending messages across each side of the bridge.
+
+Let's dive into some of these files in depth.
+
+### `contracts/Greeter.sol`
+
+There are two important functions to call out here:
+
+```javascript
+function setGreeting(string memory _greeting) public {
+  greeting = _greeting;
+  emit SetGreeting(msg.sender, tx.origin, getXorig());
+}
+```
+`setGreeting` is the function we will be calling in the other contract. It doesn't do anything other than set a greeting message.
+
+```javascript
+// Get the cross domain origin, if any
+function getXorig() private view returns (address) {
+  // Get the cross domain messenger's address each time.
+  // This is less resource intensive than writing to storage.
+  address cdmAddr = address(0);
+
+  // Mainnet
+  if (block.chainid == 1)
+    cdmAddr = 0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1;
+
+  // Kovan
```

Comment:
Guessing we could remove this given it's deprecated (although not biggie if it stays in).